docs: update with new features (#213)
Apicurio/apicurio-registry-operator: docs/modules/ROOT/partials/proc-registry-https-in-cluster.adoc

[source,bash]
----
oc create secret generic https-cert-secret --from-file=tls.key --from-file=tls.crt
----
--

. Edit the `spec.configuration.security.https` section of the `ApicurioRegistry` CR for your {registry} deployment, for example:
+
[source,yaml]
----
apiVersion: registry.apicur.io/v1
kind: ApicurioRegistry
metadata:
  name: example-apicurioregistry
spec:
  configuration:
    # ...
    security:
      https:
        secretName: https-cert-secret
----

. Verify that the connection is working:
.. Connect into a pod on the cluster using SSH (you can use the {registry} pod):
+
[source,bash]
----
oc rsh example-apicurioregistry-deployment-6f788db977-2wzpw
----

.. Find the cluster IP of the {registry} pod from the *Service* resource (see the *Location* column in the web console).
Afterwards, execute a test request (we are using self-signed certificate, so an insecure flag is required):
+
[source,bash]
----
curl -k https://172.30.230.78:8443/health
----

NOTE: In the Kubernetes secret containing the HTTPS certificate and key, the names `tls.crt` and `tls.key` must be used for the provided values.
This is currently not configurable.

.Disabling HTTP

> If you have enabled HTTPS using the procedure above, you can also disable the default HTTP connection by setting the `spec.security.https.disableHttp` to `true`.

Reviewer: ```suggestion
If you enabled HTTPS using the procedure in this section, you can also disable the default HTTP connection by setting the `spec.security.https.disableHttp` to `true`.
```

> This will remove the HTTP port 8080 from the {registry} pod container, `Service`, and the `NetworkPolicy`  (if present).
> Importantly, `Ingress` will also be removed, since the {operator} currently does not support configuring HTTPS in `Ingress`.
> Users must create an `Ingress` for HTTPS connections manually.

Reviewer: ```suggestion
This removes the HTTP port 8080 from the {registry} pod container, `Service`, and the `NetworkPolicy`  (if present).
Importantly, `Ingress` is also removed because the {operator} currently does not support configuring HTTPS in `Ingress`.
Users must create an `Ingress` for HTTPS connections manually.
```
